extensions/azurecore/src/azureResource/commands.ts

```diff
@@ -132,17 +132,25 @@ export function registerAzureResourceCommands(appContext: AppContext, tree: Azur
 
 		const selectedSubscriptionQuickPickItems = await vscode.window.showQuickPick(subscriptionQuickPickItems, { canPickMany: true });
 		if (selectedSubscriptionQuickPickItems && selectedSubscriptionQuickPickItems.length > 0) {
-			await tree.refresh(node, false);
+			for (const tree of trees) {
+				await tree.refresh(undefined, false);
+			}
 
 			selectedSubscriptions = selectedSubscriptionQuickPickItems.map((subscriptionItem) => subscriptionItem.subscription);
-			await subscriptionFilterService.saveSelectedSubscriptions(accountNode.account, selectedSubscriptions);
+			await subscriptionFilterService.saveSelectedSubscriptions(account, selectedSubscriptions);
 		}
 	});
 
-	vscode.commands.registerCommand('azure.resource.refreshall', () => tree.notifyNodeChanged(undefined));
+	vscode.commands.registerCommand('azure.resource.refreshall', () => {
+		for (const tree of trees) {
+			tree.notifyNodeChanged(undefined);
+		}
+	});
 
 	vscode.commands.registerCommand('azure.resource.refresh', async (node?: TreeNode) => {
-		await tree.refresh(node, true);
+		for (const tree of trees) {
+			await tree.refresh(node, true);
```

Comment:
We should refresh the trees in parallel

Reply:
:) I actually started with that, and then my account was throttled by azure services, LOL. same as the comment above about promise.all. I have a comment in the expandAll logic about that.

Reply:
Makes sense!